doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -455,9 +455,9 @@ Creating Hints
 
       :n:`@one_term`
         Permits declaring a hint without declaring a new
-        constant first, but this is not recommended.
+        constant first. This is deprecated.
 
-         .. warn:: Declaring arbitrary terms as hints is fragile; it is recommended to declare a toplevel constant instead
+         .. warn:: Declaring arbitrary terms as hints is fragile and deprecated; it is recommended to declare a toplevel constant instead
```

Comment:
Should this be an error rather than a warn?

Reply:
That's a documentation question that is poping up again and again (for instance recently: https://github.com/coq/coq/pull/18880#discussion_r1563169129 ). But warnings, even when set to error by default, technically remain warnings (that can be made actual warning or silenced) not errors (that cannot be bypassed).
Not sure if we should improve anything here, like having a warn_default_error category? Maybe @coq/doc-maintainers have a stronger opinion. No strong opinion on my side, I'm fine with keeping warn as we currently do (particularly considering that in most cases those are deprecations bound to disappear on next version).

Reply:
OK that was the comment that confused me in the first place. If we count these as technically warnings then I think it is fine. It would be good for some middle ground in the documentation like `warn-error` that would automatically add a sentence saying this warning is by default and error.